openedx/core/djangoapps/plugins/plugin_contexts.py

```diff
@@ -0,0 +1,81 @@
+from importlib import import_module
+
+from logging import getLogger
+
+from openedx.core.lib.cache_utils import process_cached
+
+from . import constants, registry
+
+
+log = getLogger(__name__)
+
+def get_plugins_view_context(project_type, view_name, existing_context={}):
```

Comment:
Reminder: In Python, we don't set empty dicts (`{}`) as default arguments in functions.  That's because the default value is initialized only _once_ during function declaration and so the same object would be reused across function calls.  Instead, have the default value be `None` and check for `None` in the function implementation.  See in [stackoverflow](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python).

Reply:
Good catch! That would have bit me for sure.